justin-tan/week7/wed/flickr-search/js/main.js

```diff
@@ -0,0 +1,263 @@
+console.log(`flickr search`);
+
+const FLICKR_BASE_URL = 'https://www.flickr.com/services/rest/';
+const FLICKR_API_KEY = '2f5ac274ecfac5a455f38745704ad084';
+
+let pageNum = 1
+let query = ''
+
+$( function(){
+
+    $('#query').focus(); // so we can test easily by pressing enter
+
+    // EITHER a click of the button OR (yessss) pressing enter in an input
+    // will trigger the form submit
+    $('#searchForm').on('submit', function( ev ){
+        ev.preventDefault(); // stop the form from reloading the page
+
+        query = $('#query').val();
+
+        getSearchResults( query )
+
+    }) // form submit
+
+    // next page click handler
+    $('#nextPage').on('click', function(){
+        pageNum += 1
+
+        // console.log(`button click:`, pageNum, query);
+
+        if (pageNum > 1){
+            $('#previousPage').show()
+            getPage(pageNum, query)
+        }
+    })
+
+    // previous page click handler
+    $('#previousPage').on('click', function(){
+        pageNum -= 1
+
+        if (pageNum > 1){
+            getPage(pageNum, query)
+
+        } else {
+            $('#previousPage').hide()
+            getPage(pageNum, query)
+
+        } // ifelse conditions
+    })
+
+    const pageList = numberOfPages => {
+
+        for (let i = 1; i < numberOfPages; i++) {
+            console.log(i); 
+            $('#pageList').append(`
+                <a href="#" id="${i}" class="pageTag">${i}</a>
+            `)
+        }
+    }
+    
+    $('.pageTag').on('click', function(){
+        const pageId = $('.pageTag').attr('id')
+        console.log(`page button clicked!`, pageId, query);
+        // getPage(pageId, query)
+    })
```

Comment:
doesn't work :'(?

Reply:
Got it working today! Just need to push it